firebase-perf/src/main/java/com/google/firebase/perf/metrics/HttpMetric.java

```diff
@@ -181,24 +181,21 @@ public void putAttribute(@NonNull String attribute, @NonNull String value) {
     }
   }
 
-  private void checkAttribute(@Nullable String attribute, @Nullable String value) {
+  private void checkAttribute(@Nullable String key, @Nullable String value) {
     if (isStopped) {
       throw new IllegalArgumentException(
           "HttpMetric has been logged already so unable to modify attributes");
     }
-    if (attribute == null || value == null) {
-      throw new IllegalArgumentException("Attribute must not have null key or value.");
-    }
-    if (!customAttributesMap.containsKey(attribute)
+
+    if (!customAttributesMap.containsKey(key)
         && customAttributesMap.size() >= Constants.MAX_TRACE_CUSTOM_ATTRIBUTES) {
       throw new IllegalArgumentException(
           String.format(
               Locale.ENGLISH,
               "Exceeds max limit of number of attributes - %d",
               Constants.MAX_TRACE_CUSTOM_ATTRIBUTES));
     }
-    String err =
-        PerfMetricValidator.validateAttribute(new AbstractMap.SimpleEntry<>(attribute, value));
+    String err = PerfMetricValidator.validateAttribute(new AbstractMap.SimpleEntry<>(key, value));
```

Comment:
nit: Why do we return an error instead of just throwing an IllegalArgumentException in the validate method? This makes it slightly harder to find where the exception took place.

Reply:
Like the idea. This had to be done in many parts of the code. For now, I have made this change just for the custom attributes API.